vllm/timings/utils.py

```diff
@@ -0,0 +1,49 @@
+from vllm.timings.timer import Timer
+
+__all__ = ["log_time", "get_singleton_manager"]
+
+
+def get_singleton_manager(enable_logging: bool = False):
+    """
+    Return the Timer. If not has not yet been initialized, initialize and
+    return. If it has, return the existing Timer.
+    """
+    if Timer._instance is None:
+        Timer._instance = Timer(enable_logging=enable_logging)
+    return Timer._instance
+
+
+def log_async_time(func):
+    """
+    Decorator to time async functions. Times for the function are stored using
+    the class and function names.
+    """
+
+    async def wrapper(self, *arg, **kwargs):
+        TIMER_MANAGER = get_singleton_manager()
+        func_name = f"{self.__class__.__name__}.{func.__name__}"
+        if not TIMER_MANAGER.enable_logging:
+            return await func(self, *arg, **kwargs)
+
+        with TIMER_MANAGER.time(func_name):
+            return await func(self, *arg, **kwargs)
+
+    return wrapper
+
+
+def log_time(func):
```

Comment:
I'd argue the user of this decorator shouldn't need to know there's a sync and async version. What do you think of just one decorator? Something like this might work

```python
import asyncio
from contextlib import contextmanager
import functools

...

@contextmanager
def _with_timing(func):
    TIMER_MANAGER = get_singleton_manager()
    func_name = f"{self.__class__.__name__}.{func.__name__}"
    if TIMER_MANAGER.enable_logging:
        with TIMER_MANAGER.time(func_name):
            yield
    else:
        yield


@functools.wraps(func)
def log_time(func):
    """
    Times for the function are stored using the class and function names.
    """

    def wrapper(self, *args, **kwargs):
        if not asyncio.iscoroutinefunction(func):
            with _with_timing(func):
                return func(*args, **kwargs)
        else:
            async def _async_wrapper():
                with _with_timing(func):
                    return await func(*args, **kwargs)
            return _async_wrapper()

    return wrapper
```